odl/set/pspace.py

```diff
@@ -617,13 +617,24 @@ def show(self, *args, **kwargs):
         """ Display the parts of this vector.
         """
         title = kwargs.pop('title', 'ProductSpaceVector')
-        if len(self) < 5:
-            show_parts = self.parts
-            indices = range(len(self))
+        indices = kwargs.pop('indices', None)
+
+        if indices is None:
+            if len(self) < 5:
+                indices = slice(None)
+            else:
+                indices = np.linspace(0, self.size - 1, 4, dtype=int)
         else:
-            show_parts = self.parts[:2] + self.parts[-2:]
-            indices = [0, 1, len(self) - 2, len(self) - 1]
-        for i, part in zip(indices, show_parts):
+            if isinstance(indices, tuple):
+                indices, kwargs['indices'] = indices[0], indices[1:]
+
+            if isinstance(indices, slice):
```

Comment:
Actually no, consider `indices = (slice(None), 2)`, which is valid and here we expect to unpack the first one and do stuff with it. There should be an elif on the next if however.